include/iarc7_fc_comms/CommonFcComms.hpp

```diff
@@ -202,32 +221,98 @@ FcCommsReturns CommonFcComms<T>::run()
     return flightControlImpl_.disconnect();
 }
 
-// Update flight controller status information
+// Attempt to arm the flight controller
+template<class T>
+bool CommonFcComms<T>::uavArmServiceHandler(
+                std_srvs::SetBool::Request& request,
+                std_srvs::SetBool::Response& response)
+{
+    ROS_INFO("Uav arm service handler called");
+
+    bool auto_pilot;
+    FcCommsReturns status = flightControlImpl_.isAutoPilotAllowed(auto_pilot);
+    if (status != FcCommsReturns::kReturnOk) {
+        ROS_ERROR("Failed to find out if auto pilot is enabled");
+        return false;
+    }
+
+    // If auto pilot is not enabled we have no power
+    if(!auto_pilot)
+    {
+        ROS_INFO("Failed to arm or disarm the FC: auto pilot is disabled");
+        response.success = false;
+        response.message = "disabled";
+        return true;
+    }
+
+    // Now attempt to arm or disarm
+    status = flightControlImpl_.setArm(request.data);
+    if (status != FcCommsReturns::kReturnOk)
+    {
+        ROS_ERROR("iarc7_fc_comms: Failed to send arm message");
+        return false;
+    }
+
+    // Check to see if the craft actually armed
+    ros::Time start_time = ros::Time::now();
+    while((ros::Time::now() - start_time) < ros::Duration(CommonConf::kMaxArmDelay))
+    {
+        bool armed;
+        FcCommsReturns status = flightControlImpl_.isArmed(armed);
+        if (status != FcCommsReturns::kReturnOk) {
+            ROS_ERROR("Failed to retrieve flight controller arm status");
+            return false;
```

Comment:
I think that we should continue to check. We don't know at this level if it's an unrecoverable error.